stdlib/functools.pyi

```diff
@@ -94,8 +98,17 @@ WRAPPER_ASSIGNMENTS: tuple[
 ]
 WRAPPER_UPDATES: tuple[Literal["__dict__"]]
 
-def update_wrapper(wrapper: _T, wrapped: _AnyCallable, assigned: Sequence[str] = ..., updated: Sequence[str] = ...) -> _T: ...
-def wraps(wrapped: _AnyCallable, assigned: Sequence[str] = ..., updated: Sequence[str] = ...) -> Callable[[_T], _T]: ...
+class _Wrapped(Generic[_P1, _R1, _P2, _R2]):
+    __wrapped__: Callable[_P2, _R2]
+    def __call__(self, *args: _P1.args, **kwargs: _P1.kwargs) -> _R2: ...
```

Comment:
On the other hand, `update_wrapper()` doesn't change the wrapper's signature (unless I'm missing something):

```python
from functools import update_wrapper

def wrapper(x: int) -> str:
    return str(x)

def wrapped(y: str, z: bool) -> float:
    return 42.0

wrapper2 = update_wrapper(wrapper, wrapped)

assert wrapper2 is wrapper
print(wrapper(13))
print(wrapper("", True))  # TypeError
```

Same for `wraps()`:

```python
from functools import wraps

def my_decorator(f):
    @wraps(f)
    def wrapper() -> int:
        return 42
    return wrapper

@my_decorator
def wrapped(x: int) -> bool:
    return True

print(type(wrapped()))  # -> class 'int'
print(type(wrapped(42)))  # TypeError
```

So this should read:

```suggestion
    def __call__(self, *args: _P2.args, **kwargs: _P2.kwargs) -> _R2: ...
```

Reply:
Good catch. Yes, after looking at your examples above, I realize that I had the ParamSpecs swapped in `_Wrapped`. Now that I've swizzled those, your recommendation for renaming the ParamSpecs makes sense.